test/old/testdir/test_filetype.vim

```diff
@@ -94,7 +118,7 @@ let s:filename_checks = {
     \ 'bzr': ['bzr_log.any', 'bzr_log.file'],
     \ 'c': ['enlightenment/file.cfg', 'file.qc', 'file.c', 'some-enlightenment/file.cfg'],
     \ 'cabal': ['file.cabal'],
-    \ 'cabalconfig': ['cabal.config'],
+    \ 'cabalconfig': ['cabal.config', expand("$HOME/.config/cabal/config")] + WhenConfigHome('$XDG_CONFIG_HOME/cabal/config'),
```

Comment:
The vim9 script version of the file gets this dict via calling`s:GetFilenameChecks()`, which is evaluated when `CheckItems()` is called.

Because we define this at the file scope in the script, it only gets evaluated when the script is sourced, but the logic for `s:SetupConfigHome()` isn't run yet (so `$XDG_CONFIG_HOME` is empty), which is probably why the tests are failing.

Reply:
The following diff seems to make the tests pass locally for me:

```diff
diff --git a/test/old/testdir/test_filetype.vim b/test/old/testdir/test_filetype.vim
index ad243e424..f54ec62f8 100644
--- a/test/old/testdir/test_filetype.vim
+++ b/test/old/testdir/test_filetype.vim
@@ -66,7 +66,8 @@ endfunc
 
 " Filetypes detected just from matching the file name.
 " First one is checking that these files have no filetype.
-let s:filename_checks = {
+func s:GetFilenameChecks() abort
+  return {
     \ 'none': ['bsd', 'some-bsd'],
     \ '8th': ['file.8th'],
     \ 'a2ps': ['/etc/a2ps.cfg', '/etc/a2ps/file.cfg', 'a2psrc', '.a2psrc', 'any/etc/a2ps.cfg', 'any/etc/a2ps/file.cfg'],
@@ -725,6 +726,7 @@ let s:filename_checks = {
     \
     \ 'help': [$VIMRUNTIME . '/doc/help.txt'],
     \ }
+endfunc
 
 let s:filename_case_checks = {
     \ 'modula2': ['file.DEF'],
@@ -762,7 +764,7 @@ func Test_filetype_detection()
   call mkdir(GetConfigHome(), 'R')
 
   filetype on
-  call CheckItems(s:filename_checks)
+  call CheckItems(s:GetFilenameChecks())
   if has('fname_case')
     call CheckItems(s:filename_case_checks)
   endif
```